ocf_data_sampler/load/load_dataset.py

```diff
@@ -0,0 +1,55 @@
+""" Loads all data sources """
+import xarray as xr
+
+from ocf_data_sampler.config import Configuration
+from ocf_data_sampler.load.gsp import open_gsp
+from ocf_data_sampler.load.nwp import open_nwp
+from ocf_data_sampler.load.satellite import open_sat_data
+from ocf_data_sampler.load.site import open_site
+
+
+def get_dataset_dict(config: Configuration) -> dict[xr.DataArray, dict[xr.DataArray]]:
+    """Construct dictionary of all of the input data sources
+
+    Args:
+        config: Configuration file
+    """
+
+    in_config = config.input_data
+
+    datasets_dict = {}
+
+    # Load GSP data unless the path is None
+    if in_config.gsp and in_config.gsp.gsp_zarr_path:
```

Comment:
I think checking for `in_config.gsp` is enough. `zarr_path` is a required field so config model should be checking it exists for you

Reply:
Yea, ok, need to look into this. There's defiantely a test that makes sure it works with no gsp data. Ill have a further look at this

Reply:
Yea, there is a funny edge case, where in live we have no live GSP data, as it's not used for the forecast. But we still need `gsp.time_resolution_minutes`, `gsp_config.forecast_minutes` and `gsp_config.history_minutes` for the sun coords position. There's a slight case for making a Sun configuration but, perhaps its better in a different PR